### DISCUSSION TITLE
randomly skipping a trial within a timeline?

### DISCUSSION THREAD

**Discussion by sambfloyd** [jspsych/jsPsych#967] [Q&A] [answered]
hi! I'm wondering what a good way to randomly skip a trial within a timeline is (~50% of the time, I want my timeline to skip one of my trials). 

I am currently having the code set the trial_duration to 0ms ~50% of the time, which works great, except that my task is for mobile, and the audio on the subsequent trial won't automatically play without a user action in the preceding trial (which isn't possible if the trial_duration is 0ms).

I'm hoping that there's something like the conditional_function parameter but for individual trials, and I just didn't see it anywhere?

Not sure code would be helpful here but I can add it. Thanks so much for jspsych and for this community!!!

**Comment by becky-gilbert** [comment, marked as answer]
Hi,
I'm not totally sure I understand what you're trying to do. Are you using a trial procedure with timeline variables, or are you setting up your trials in a different way? 
Maybe you could put your trial object(s) into a conditional node? For instance, with timeline variables it might look like this:
```javascript
var trial_conditional = {
  timeline: [trial]
  conditional_function: function() {
    // code here that checks if the trial should run
    // and returns true if so, or false if it should be skipped
  }
}
var trial_procedure = {
  timeline: [trial_conditional],
  timeline_variables: [{...}]
}
```
